test/feature/AsyncGenerators/ForOn.js

```diff
@@ -0,0 +1,98 @@
+// Async.
+// Options: --async-functions --for-on
+
+// TODO(mnieper) Many more tests.
+
+class Observable {
+  constructor(observe) {
+    this[Symbol.observer] = observe;
+  }
+}
+
+function decorate(generator, onDone) {
+  var decoratedGenerator = Object.create(generator);
+  decoratedGenerator.next = function(v) {
+    var result = generator.next(v);
+    if (result && result.done) {
+      onDone();
+    }
+    return result;
+  };
+
+  ["throw","return"].forEach(method => {
```

Comment:
LOL

The end user story is pretty much here: https://github.com/jhusain/asyncgenerator